Debian 8: support git installation on non-x86_64 architectures

vutny opened the pull request:
### What does this PR do?
- Fixes Salt **git** installation on Debian 8 `i386` (by adding backports repo)
- If `pip` installation allowed, always grab latest `tornado` Python module from there
- Installs `python-tornado` package from SaltStack corp repo on `x86_64` if `pip` not allowed
### What issues does this PR fix or reference?

This PR should resolve #774 by allowing git installation on non-`x86_64` architectures such as `arm*`.

Example usage:

``` sh
sh bootstrap-salt.sh -D -r -P git v2016.3.0
```
### Previous Behavior
- Fail to bootstrap Salt from git on non-`x86_64` architecture
### New Behavior
- Display error message on non-`x86` architectures with suggestion how to perform git installation mode with disabled repo configuration.
### Tests written?

No. Tested on Debian 8.4 `x86_64` and `i386` systems.

vutny committed:
Debian 8: support git installation on non-x86 architectures with `-r`

vutny committed:
Debian 8 git: do not configure backports repo when using `-r`

vutny commented:
Updated description.

vutny committed:
Re-format usage output to fix 80 characters width terminal

vutny committed:
Debian: add note to README about supported system architectures

rallytime commented:
Thanks @vutny. This looks good to me, but I'd like @s0undt3ch to look at this one as well.

vutny commented:
@rallytime Thank you for taking notice on my PR!

vutny committed:
Detect OS packages architecture on Debian instead of relying on CPU data

vutny committed:
Debian: add suggestion to try git installation mode on i386 arch

vutny committed:
Debian: add python-psutil to deps (required for 2016.3)

vutny committed:
Create default logs directory which is absent after git installations

vutny committed:
Fix `-X` and `-d` options to do what they exactly supposed to do

vutny committed:
Debian: install majority of Salt dependencies for git mode only.
        Apt should resolve and install all deps for Salt packages

rallytime commented:
Hi @vutny - It looks like this will need to be rebased before we can get it in as there are some merge conflicts now.

vutny committed:
Set STABLE_REV variable to "latest" early, properly fixes git bootstraps

vutny committed:
Handle `armhf` arch SaltStack repository for Debian/Raspbian

vutny commented:
Hi @rallytime - Thanks for keeping an eye on this!
Now rebased and revised.

There are many changes, I know. Most of it touches git installation on Debian, but also I've fixed few generic issues I found in the process. I guess my PR could possibly supersede some earlier changes which attempted to address the same issues.

Anyhow, I've tried to test it in all the ways I could. Looking forward for this PR to be merged.
Thank you.

vutny committed:
Update README about supported architectures on Debian 8

rallytime commented:
@vutny I'd like to just get this in, as it has some good improvements!

In the future, I would to see changes like this split up into multiple pull requests, instead of one large PR that has a lot of refactoring, plus doc changes, and other things. It's much much easier to review smaller, discrete changes. Otherwise, this looks great. Thank you for your work here!

vutny commented:
Thank you, @rallytime !
I understand the drawbacks of such complex changes to be accepted at once. Will do more granular changes in the future.

rallytime commented:
Thanks again @vutny!
